diff --git a/cmd/nvedit.c b/cmd/nvedit.c
index 1cb0bc14..8f1c4015 100644
--- a/cmd/nvedit.c
+++ b/cmd/nvedit.c
@@ -358,6 +358,7 @@ ulong env_get_hex(const char *varname, ulong default_val)
 	return value;
 }
 
+#ifdef CONFIG_NET
 int eth_env_get_enetaddr(const char *name, uint8_t *enetaddr)
 {
 	eth_parse_enetaddr(env_get(name), enetaddr);
@@ -375,6 +376,7 @@ int eth_env_set_enetaddr(const char *name, const uint8_t *enetaddr)
 
 	return env_set(name, buf);
 }
+#endif
 
 #ifndef CONFIG_SPL_BUILD
 static int do_env_set(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
